### PR TITLE
fix(menu): do not check right from selfsrevice

### PR DESCRIPTION
**Pull request by stonebuzz**
Add icon menu entry

Do not check ```plugin_glpiinventory_selfpackage``` ```READ``` right from ```helpdesk```

Because GLPI clean ```$_SESSION['glpiactiveprofile']```to keep only 

```php
   /// Helpdesk fields of helpdesk profiles
    public static $helpdesk_rights = [
        'create_ticket_on_login',
        'changetemplates_id',
        'followup',
        'helpdesk_hardware',
        'helpdesk_item_type',
        'knowbase',
        'password_update',
        'personalization',
        'problemtemplates_id',
        'reminder_public',
        'reservation',
        'rssfeed_public',
        'show_group_hardware',
        'task',
        'ticket',
        'ticket_cost',
        'ticket_status',
        'tickettemplates_id',
        'ticketvalidation',
    ];
```
If current user have any package to deploy, menu entry is not added.
See : https://github.com/glpi-project/glpi-inventory-plugin/blob/21e1c540edd90c223b92d5f8a7d6a93ebaf21c50/setup.php#L311-L322

### CODE REVIEW

**Review by trasher — APPROVED** [review]

File: front/deploypackage.public.php [hunk 1]
```diff
@@ -34,7 +34,9 @@
 include("../../../inc/includes.php");
 Session::checkLoginUser();
 
-Session::checkRight('plugin_glpiinventory_selfpackage', READ);
+if (!Session::getCurrentInterface() == 'helpdesk') {
```
Comment (cedric-anne) [inline front/deploypackage.public.php:37]:
```suggestion
if (Session::getCurrentInterface() !== 'helpdesk') {
```

File: inc/deploypackage.class.php [hunk 2]
```diff
@@ -2055,8 +2055,10 @@ public function canUserDeploySelf()
     {
         global $DB;
 
-        if (!Session::haveRight('plugin_glpiinventory_selfpackage', READ)) {
-            return false;
+        if (!Session::getCurrentInterface() == 'helpdesk') {
```
Comment (cedric-anne) [inline inc/deploypackage.class.php:2058]:
```suggestion
        if (Session::getCurrentInterface() !== 'helpdesk') {
```

**Review by cedric-anne — COMMENTED** [review]
Is this secure to not check this right for all helpdesk users ?

**Comment by cedric-anne** [comment]
Is this secure to not check this right for all helpdesk users ?

**Review by cedric-anne — DISMISSED** [review]

**Comment by cedric-anne** [comment]
> Is this secure to not check this right for all helpdesk users ?

@stonebuzz As far as I know, it may permit user to deploy packages they are not supposed to.

**Comment by stonebuzz** [comment]
> > Is this secure to not check this right for all helpdesk users ?
> 
> @stonebuzz As far as I know, it may permit user to deploy packages they are not supposed to.

No it allows the user to launch deploy packages if there is one available for that user (computer)

Is no one are available, menu entry is not displayed

**Comment by trasher** [comment]
ping @cedric-anne

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**